packages/server/lib/controllers/v1/account/signup.ts

```diff
@@ -63,30 +61,64 @@ export const signup = asyncWrapper<Signup>(async (req, res) => {
         return;
     }
 
-    const account = await accountService.createAccount(`${name}'s Organization`);
+    let account: DBTeam | null;
+    if (token) {
+        // Invitation signup
+        const validToken = await getInvitation(token);
+        if (!validToken) {
+            res.status(400).send({ error: { code: 'invalid_invite_token', message: 'The token used was found to be invalid.' } });
+            return;
+        }
 
-    if (!account) {
-        logger.error('Error creating account');
-        res.status(500).send({ error: { code: 'error_creating_account', message: 'There was a problem creating the account. Please reach out to support.' } });
-        return;
+        account = await accountService.getAccountById(validToken.account_id);
+        if (!account) {
+            res.status(500).send({ error: { code: 'server_error', message: 'Failed to get team' } });
+            return;
+        }
+        void analytics.track(AnalyticsTypes.ACCOUNT_JOINED, account.id, {}, isCloud ? { email } : {});
+
+        await acceptInvitation(token);
+    } else {
+        // Regular account
+        account = await accountService.createAccount(`${name}'s Organization`);
+        if (!account) {
+            res.status(500).send({
+                error: { code: 'error_creating_account', message: 'There was a problem creating the account. Please reach out to support.' }
+            });
+            return;
+        }
     }
 
+    // Create user
     const salt = crypto.randomBytes(16).toString('base64');
     const hashedPassword = (await util.promisify(crypto.pbkdf2)(password, salt, 310000, 32, 'sha256')).toString('base64');
     const user = await userService.createUser(email, name, hashedPassword, salt, account.id, false);
-
     if (!user) {
-        logger.error('Error creating user');
         res.status(500).send({ error: { code: 'error_creating_user', message: 'There was a problem creating the user. Please reach out to support.' } });
         return;
     }
 
-    if (!user.email_verification_token) {
-        res.status(400).send({ error: { code: 'email_already_verified', message: 'Email address was already verified, please login.' } });
+    // Ask for email validation if not coming from an invitation
+    if (!token) {
```

Comment:
I  would have inverted to avoid the negation but I am nitpicking ;)
```
// log directly if we are coming from an invite
if(token) {
   ...
   return
}

// Ask for email validation if not coming from an invitation
...

Reply:
changed 👌🏻